erigon-lib/state/aggregator2.go

```diff
@@ -279,3 +287,33 @@ func EnableHistoricalRCache() {
 }
 
 var ExperimentalConcurrentCommitment = false // set true to use concurrent commitment by default
+
+func Compatibility(d datadir.Dirs) error {
+	directories := []string{
+		d.Chaindata, d.Tmp, d.SnapIdx, d.SnapHistory, d.SnapDomain,
+		d.SnapAccessors, d.SnapCaplin, d.Downloader, d.TxPool, d.Snap,
+		d.Nodes, d.CaplinBlobs, d.CaplinIndexing, d.CaplinLatest, d.CaplinGenesis,
+	}
+	for _, dirPath := range directories {
+		err := filepath.WalkDir(dirPath, func(path string, entry fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if !entry.IsDir() {
+				name := entry.Name()
+				if strings.HasPrefix(name, "v1.0-") {
```

Comment:
1. Create bool variable with meaningful name and assign there logic result of 
```
if !entry.IsDir() {
 				name := entry.Name()
 				if strings.HasPrefix(name, "v1.0-") {
```

2. Unclear what if user has `v2.0`? In this case we should not recommend `erigon seg reset-to-old-ver-format`